### ISSUE TITLE
Segfault with dio_write and zend_fetch_resource

### ISSUE THREAD

**Issue by xor-gate** [mhei/pecl_dio#2]
Hi @mhei,

I was porting myself the pecl dio module to php7 and failed because it kept segfaulting. Then I googled around and found you also ported it. With my setup it seems to fail.

*loopback.php*
```
#!/usr/bin/php
<?php

print_r(get_defined_constants(true)['dio']);

$fd = dio_open('/dev/ttyUSB0', O_RDWR);
debug_zval_dump($fd);
//dio_tcsetattr($fd, ['baud' => 9600, 'bits' => 8, 'stop' => 1, 'parity' => 0]);

while (true) {
	dio_write($fd, 'U');
	$r = dio_read($fd, 1);
	print $r . PHP_EOL;
	sleep(1);
}

dio_close($fd);
```

```
Starting program: /usr/bin/php7.0 -n -dextension=modules/dio.so loopback.php
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Array
(
~~~ SNIP ~~~
    [O_RDWR] => 2
~~~ SNIP ~~~
)
resource(4) of type (Direct I/O File Descriptor) refcount(2)

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff5bfac68 in zif_dio_write (execute_data=<optimized out>, return_value=0x7ffff5e13110) at /home/jjacobs/repos/github/pecl_dio/dio.c:245
245		if ((f = (php_fd_t *)zend_fetch_resource(Z_RES_P(r_fd), le_fd_name, le_fd)) == NULL) {
(gdb) f 0
#0  0x00007ffff5bfac68 in zif_dio_write (execute_data=<optimized out>, return_value=0x7ffff5e13110) at /home/jjacobs/repos/github/pecl_dio/dio.c:245
245		if ((f = (php_fd_t *)zend_fetch_resource(Z_RES_P(r_fd), le_fd_name, le_fd)) == NULL) {
(gdb) p r_fd
$1 = (zval *) 0x7fff00000000
```

This looks weird, as the lower part of the pointer is zero. 

My system info:
```
Debian 8 AMD64
PHP7 from DotDeb -> https://www.dotdeb.org/category/php/
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012

PHP 7.0.12-1~dotdeb+8.1 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.12-1~dotdeb+8.1, Copyright (c) 1999-2016, by Zend Technologies
```

I test against a FTDI FT232 USB serial port (ttyUSB0) with a loopback cable.

**Comment by xor-gate** [comment]
It took some deep digging, I tried with a named fifo which worked sort-of. I needed to patch to make it actually work:

https://github.com/dualinventive/pecl_dio/commit/52e1ff1f8fd06d362945f34ce785b96683f7630d

I followed some conventions from: https://wiki.php.net/phpng-upgrading

**Comment by mhei** [comment]
Your patch looks good to me - only one nitpick: I would remove the baudrate change because its unrelated to the segfault error. I've pushed my "OpenWrt/LED patch series creating" branch which already contains an additional patch for adding more baudrates...

**Comment by xor-gate** [comment]
You are right, this where some leftovers from testing. Also the flow control should be disabled by default imho.

**Comment by xor-gate** [comment]
Would you like to receive a clean pull request?

**Comment by mhei** [comment]
Sorry, I was traveling with no good internet access. A clean pull request would be fine, thanks.

**Comment by xor-gate** [comment]
No problem, thank you for your response. I have created PR #3. And removed the added baudrate and reverted the `_args list` argument for the php functions.

**Comment by foufou1** [comment]
Fatal error: Cannot use 'Int' as class name as it is reserved in C:\Server\Webdev\Exemple\library\Zend\Filter\Int.php on line 12 
someone help me please

**Comment by mhei** [comment]
I'm closing this issue and the PR because upstream released a version 0.0.9 where this issue should be fixed.
